wapi/session.py

```diff
@@ -468,6 +462,23 @@ def data_request(self, req_type, urlbase, url, data=None, rawdata=None, authval=
             else:
                 self.auth.validate_auth()
                 headers.update(self.auth.get_headers(databytes))
+        
+        return headers
+    
+    def send_data_request(self, req_type, urlbase, url, data=None, rawdata=None, headers=None, authval=None,
```

Comment:
I think the naming is a bit confusing: `send_data_request` vs `data_request`. It looks like there is the important distinction between making a request with client-id and client-secret and requests with token. If we end up with some functions `request_with_client_credentials` and `request_with_token` (or nicer naming 😉 ), then this will make it easier to understand. This will also reduce the amount of optional params per function, I hope.

Reply:
Yes there should be a better name than `send_data_request`, so I am open to suggestions here 😄.

It is true that requests sent to auth server would use client credentials and requests sent to wapi would use token as headers. But I can't see an obvious way to distinct them into 2 functions and make their signatures less messy, e.g the `request_with_client_credentials` might still need headers in their parameters list in future usecase. Are you ok with addressing `data_request` refactoring in another PR, and leave this PR for the purpose of fixing the bug and consolidating the tests as it is now?